resolver/resolver.cc

```diff
@@ -1842,6 +1840,7 @@ class ResolveSignaturesWalk {
     }
 
     unique_ptr<ast::Expression> postTransformMethodDef(core::Context ctx, unique_ptr<ast::MethodDef> original) {
+        deleteOptionlArg(ctx, original.get());
```

Comment:
It seems that this will break IDE features on the default argument as well as typing for it.
Seems to be a step back to me

Reply:
Well almost all defaults are currently erased, this is just catching this weird case. Would you prefer they all stay in the AST? Is there a test somewhere that can show the feature you're trying to keep (nothing is failing so I assumed this didn't change something for the users that we cared about).